core/src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -483,6 +485,20 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
+        if (from == null && to == null) {
+            /**
+             * Open bounds on both side, we can rewrite to an exists query
+             * if the {@link FieldNamesFieldMapper} is enabled.
+             */
+            final FieldNamesFieldMapper.FieldNamesFieldType fieldNamesFieldType =
+                (FieldNamesFieldMapper.FieldNamesFieldType) context.getMapperService().fullName(FieldNamesFieldMapper.NAME);
+            if (fieldNamesFieldType == null) {
+                return new MatchNoDocsQuery("No mappings yet");
+            }
+            if (fieldNamesFieldType.isEnabled()) {
+                return ExistsQueryBuilder.newFilter(context, fieldName);
```

Comment:
maybe leave a comment that we need to check whether the fieldNames mapper is enabled because otherwise we fail exists queries